fish-rust/src/event.rs

```diff
@@ -491,7 +498,8 @@ struct PendingSignals {
 impl PendingSignals {
     /// Mark a signal as pending. This may be called from a signal handler. We expect only one
     /// signal handler to execute at once. Also note that these may be coalesced.
-    pub fn mark(&self, which: usize) {
+    pub fn mark(&self, sig: Signal) {
```

Comment:
As this is run within a signal handler, it should continue to take a `usize` to ensure it is easy to reason about.